plugin/user.c

```diff
@@ -100,19 +103,184 @@ static int load_config_user(void) {
   return 0;
 }
 
+static tai_hook_ref_t g_sceRegMgrGetKeyInt_SceSystemSettingsCore_hook;
+static int sceRegMgrGetKeyInt_SceSystemSettingsCore_patched(const char *category, const char *name, int *value) {
+  if (sceClibStrncmp(category, "/CONFIG/HENKAKU", 15) == 0) {
+    if (value) {
+      load_config_user();
+      if (sceClibStrncmp(name, "enable_psn_spoofing", 19) == 0) {
+        *value = config.use_psn_spoofing;
+      } else if (sceClibStrncmp(name, "enable_unsafe_homebrew", 22) == 0) {
+        *value = config.allow_unsafe_hb;
+      } else if (sceClibStrncmp(name, "enable_version_spoofing", 23) == 0) {
+        *value = config.use_spoofed_version;
+      }
+    }
+    return 0;
+  }
+
+  return TAI_CONTINUE(int, g_sceRegMgrGetKeyInt_SceSystemSettingsCore_hook, category, name, value);
+}
+
+static tai_hook_ref_t g_sceRegMgrSetKeyInt_SceSystemSettingsCore_hook;
+static int sceRegMgrSetKeyInt_SceSystemSettingsCore_patched(const char *category, const char *name, int value) {
+  if (sceClibStrncmp(category, "/CONFIG/HENKAKU", 15) == 0) {
+    if (sceClibStrncmp(name, "enable_psn_spoofing", 19) == 0) {
+      config.use_psn_spoofing = value;
+    } else if (sceClibStrncmp(name, "enable_unsafe_homebrew", 22) == 0) {
+      config.allow_unsafe_hb = value;
+    } else if (sceClibStrncmp(name, "enable_version_spoofing", 23) == 0) {
+      config.use_spoofed_version = value;
+    }
+    save_config_user();
+    henkaku_reload_config();
+    return 0;
+  }
+
+  return TAI_CONTINUE(int, g_sceRegMgrSetKeyInt_SceSystemSettingsCore_hook, category, name, value);
+}
+
+static int build_version_string(int version, char *string, int length) {
+  char a = (version >> 24) & 0xF;
+  char b = (version >> 20) & 0xF;
+  char c = (version >> 16) & 0xF;
+  char d = (version >> 12) & 0xF;
+  if (a || b || c || d) {
```

Comment:
maybe `if (version)` is better? I'm not sure if compiler optimizations can remove the 4 conditional checks.